x-pack/test/functional/services/ml/data_visualizer_table.ts

```diff
@@ -133,17 +153,119 @@ export function MachineLearningDataVisualizerTableProvider(
       );
     }
 
-    public async assertViewInLensActionEnabled(fieldName: string) {
+    public async ensureActionsMenuOpen(fieldName: string) {
+      await retry.tryForTime(30 * 1000, async () => {
+        const panel = await find.existsByCssSelector('euiContextMenuPanel');
+        if (!panel) {
+          await testSubjects.click(this.rowSelector(fieldName, 'euiCollapsedItemActionsButton'));
+          await find.existsByCssSelector('euiContextMenuPanel');
+        }
+      });
+    }
+
+    public async ensureActionsMenuClosed(fieldName: string, action: string) {
+      await retry.tryForTime(30 * 1000, async () => {
+        if (await testSubjects.exists(action)) {
+          await testSubjects.click(this.rowSelector(fieldName, 'euiCollapsedItemActionsButton'));
+          await testSubjects.missingOrFail(action, { timeout: 5000 });
+        }
+      });
+    }
+
+    public async assertActionMenuViewInLensEnabled(fieldName: string, expectedValue: boolean) {
+      await retry.tryForTime(30 * 1000, async () => {
+        await this.ensureActionsMenuOpen(fieldName);
+        const actionMenuViewInLensButton = await find.byCssSelector(
+          '[data-test-subj="dataVisualizerActionViewInLensButton"][class="euiContextMenuItem"]'
+        );
+        const isEnabled = await actionMenuViewInLensButton.isEnabled();
+        expect(isEnabled).to.eql(
+          expectedValue,
+          `Expected "Explore in lens" action menu button for '${fieldName}' to be '${
+            expectedValue ? 'enabled' : 'disabled'
+          }' (got '${isEnabled ? 'enabled' : 'disabled'}')`
+        );
+      });
+    }
+
+    public async assertActionMenuDeleteIndexPatternFieldButtonEnabled(
+      fieldName: string,
+      expectedValue: boolean
+    ) {
+      await this.ensureActionsMenuOpen(fieldName);
+      const actionMenuViewInLensButton = await find.byCssSelector(
+        '[data-test-subj="dataVisualizerActionDeleteIndexPatternFieldButton"][class="euiContextMenuItem"]'
+      );
+      const isEnabled = await actionMenuViewInLensButton.isEnabled();
+      expect(isEnabled).to.eql(
+        expectedValue,
+        `Expected "Delete index pattern field" action menu button for '${fieldName}' to be '${
+          expectedValue ? 'enabled' : 'disabled'
+        }' (got '${isEnabled ? 'enabled' : 'disabled'}')`
+      );
+    }
+    public async clickActionMenuDeleteIndexPatternFieldButton(fieldName: string) {
+      const testSubj = 'dataVisualizerActionDeleteIndexPatternFieldButton';
+      await retry.tryForTime(5000, async () => {
+        await this.ensureActionsMenuOpen(fieldName);
+
+        const button = await find.byCssSelector(
+          '[data-test-subj="dataVisualizerActionDeleteIndexPatternFieldButton"][class="euiContextMenuItem"]'
+        );
+        await button.click();
+        await this.ensureActionsMenuClosed(fieldName, testSubj);
```

Comment:
Actually, I think we would like to assert that the action menu is closed.
Currently with `ensureActionsMenuClosed`, we're checking if it's still there and try to close it if needed. This could cause isssue as at this point there's the delete confirm modal open. So I'd suggest to either change this to a simple assertion for the missing/closed action menu or make it instead assert that the delete confirmation model is open.

Reply:
Updated here https://github.com/elastic/kibana/pull/101316/commits/6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e